diwako_cbrn.vr/scripts/cbrn/functions/fn_detectorGeigerPFH.sqf

```diff
@@ -0,0 +1,35 @@
+if (!isGameFocused || isGamePaused) exitWith {};
+params ["_args", "_idPFH"];
+_args params ["_lastIteration"];
+
+private _time = cba_missiontime;
+private _delta = _time - _lastIteration;
+_args set [0, _time];
+
+if ( !cbrn_geiger ||  {!(ace_player getVariable ["cbrn_detector_geiger", false])} ) exitWith {
+    cbrn_geigerPfh = -1;
+    [_idPFH] call CBA_fnc_removePerFrameHandler;
+};
+
+if (cbrn_curThreat < 0.25) exitWith {};
+
+private _wait = 2;
+
+if ((cbrn_lastGeiger + _wait) <= _time) then {
+    cbrn_lastGeiger = _time;
+	deleteVehicle _soundSrc;
+	
+	private _soundPlayed = "";
+	
+	if (cbrn_curThreat > 0.25) then { _soundPlayed = "cbrn_low_rad"; };
+	if (cbrn_curThreat > 0.75) then { _soundPlayed = "cbrn_mid_rad"; };
+	if (cbrn_curThreat > 1.25) then { _soundPlayed = "cbrn_high_rad"; };
+	if (cbrn_curThreat > 1.75) then { _soundPlayed = "cbrn_danger_rad"; };
+	if (cbrn_curThreat > 2.25) then { _soundPlayed = "cbrn_deadly_rad"; };
+	if (cbrn_curThreat > 2.75) then { _soundPlayed = "cbrn_run_rad"; };
+	if (cbrn_curThreat > 3.25) then { _soundPlayed = "cbrn_omg_rad"; };
+	
+	ace_player setVariable ["cbrn_last_sound", _soundPlayed];
+
+	private _soundSrc = ace_player say3D _soundPlayed;
```

Comment:
`_soundSource` written and not used afterwards. Another thing to think about is the usage of `say3D`. What is the benefit of using it in 3D space if the origin is literally yourself?

Reply:
For the Geiger counter I wanted to specifically play the sound in 3D so others could hear it as well; I think there is definite value in having a family of sound-makers that are able to be heard by others.

Opens up some interesting situations as well, if the guy holding your counter goes down, for example.

Reply:
hm, say3D is a local command. In MP others will not hear it, if it is implemented as such. You would need to `remoteExec` this, but `remoteExec` has the issue with battle eye enabled servers.

So first off how often would you send a network message when a sound plays? from there we can think about how to sync this in mp.

Reply:
I didn't know that say3D was local. I'll get working on the other points and ask a few people if they have an idea on how to approach this.

Reply:
The approach is fairly easy. A CBA event just for doing `say3D` on a given object with say3D params.

I also just noticed that I am using remoteExec for the low oxygen beeps. So you can throw that out of the window anyways, lol.  You can do it like this
https://github.com/diwako/diwako_cbrn/blob/800ff3d43fdb915ab86cd1bd5bd00a3bc57d7aed/diwako_cbrn.vr/scripts/cbrn/functions/fn_startOxygen.sqf#L73

The question remains how often a sound is being played here. It should not be constant sound going on each second. That would be too much network traffic on the long run.

Reply:
Hm, another idea would be to use `playSound3D`. Tho not sure due to the length of the sound

Reply:
Yeah, I'll capitulate and switch over to playSound3D. As you said, the length of the sound is going to be an issue, but I think I'll slice the bites down into half a second sounds - if this doesn't cause the filesize to explode. This solution kinda sucks, but hey, so does ArmA sometimes 🙃

Reply:
It is also possible to make the sounds play locally, you'd however need to synchronize the intensity that should be played.

e.g.
Someone selects to enable the geiger counter. it will send a global event (none jip) to everyone. Clients with an interface have this event added to an event handler.

It will then check a value on the unit, best just an integer and from there it will just play the sound according to it, until the person what holds said geiger counter turns it off again. then the loop ends itself.

for the synched value, it can be just the cbrn_curThreat value. of course only synced when different value and if the geiger counter is used

Reply:
Alright, so;

I solved the problem by switching back to playSound3D but using significantly shorter soundfiles; they're .5 secs bits out of 2 sec long files; They are nyumbered from 0->3 and are played sequentially in that order.

This solves the problem of trying to sync the position to the player; it's far less noticeable if someone sprints off and easier to call and synchronize; the downside is that the files require a little more space (roughly 100kb more compared to the full sound files) which should be negligible in the greater scheme of things.